### PR TITLE
Eth and net rpc funcs, smart contract abstraction

### PR DESCRIPTION
**Pull request by pabelnl**

### CODE REVIEW

File: pocket-aion/AionContract.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                    }
+                    handler([resultStr],nil)
+                }
+            })
+        }
+        
+    }
+    
+    // MARK: Tools
+    public static func encodeFunctionCall(function: Function, params: [Any]) throws -> String{
```
Comment (luyzdeleon) [inline pocket-aion/AionContract.swift]: Make this private?

File: pocket-aion/AionContract.swift [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        for abiElement in abiDefinition {
+            guard let function = try Function.parseFunctionElement(functionJSON: abiElement) else {
+                return
+            }
+        
+            functions.append(function)
+        }
+    }
+
+    public func getFunctionFromArray(name: String, functions: [Function]) -> Function?{
```
Comment (luyzdeleon) [inline pocket-aion/AionContract.swift]: Make this private.

File: pocket-aion/AionContract.swift [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            guard let result = jsonObj[key]?.string else{
+                return nil
+            }
+            
+            return result
+        }
+        return nil
+    }
+    
+    public func getArrayByKeyFromJSONArray(key: String, jsonArray: [JSON]) -> [JSON]?{
```
Comment (luyzdeleon) [inline pocket-aion/AionContract.swift]: I think this function alongside your other JSON helper functions could be in their util struct.

File: pocket-aion/PocketAion.swift [hunk 4]
```diff
@@ -167,6 +174,34 @@ public class PocketAion: Pocket, PocketPlugin {
         return pocketTx
     }
     
+    // TODO MAKE PRIVATE BI WAW WA WA
```
Comment (luyzdeleon) [inline pocket-aion/PocketAion.swift]: Remove comment?

File: pocket-aion/PocketAion.swift [hunk 5]
```diff
@@ -167,6 +174,34 @@ public class PocketAion: Pocket, PocketPlugin {
...
+    public static func encodeFunction(functionStr: String, params: String) throws -> String{
```
Comment (luyzdeleon) [inline pocket-aion/PocketAion.swift]: Make private?

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

File: pocket-aion/AionContract.swift (hunk 1 above)
Comment (pabelnl) [reply pocket-aion/AionContract.swift]: doing this give error on the method doing the call to AionContract.getEncodedFunctionCall in the Function struct

File: pocket-aion/PocketAion.swift (hunk 4 above)
Comment (pabelnl) [reply pocket-aion/PocketAion.swift]: my bad 💃

**Pull request merged**